### PR TITLE
ci: Run more things in parallel and only when needed

### PR DESCRIPTION
**Pull request by larseggert**
Similar to what was done for neqo.

Also address a spurious test failure when a privileged port was randomly chosen (since I ran a lot of tests during debugging I managed to hit that.)

TODO: Figure out why the `windows_bindgen` tests only succeeds when run via `llvm-cov`. That seems odd.

### CODE REVIEW

**Comment by larseggert** [comment]
@djc any idea why what `windows_bindgen` test is now failing on some but not all Windows targets?

**Comment by larseggert** [comment]
@djc It looks like `codegen_windows_bindings` **only** succeeds when run via `llvm-cov`?!

File: src/lib.rs [hunk 1]
```diff
@@ -340,8 +341,8 @@ mod test {
                 // We found an unused port.
                 Ok(socket) => return socket.local_addr().unwrap(),
                 Err(e) => match e.kind() {
-                    std::io::ErrorKind::AddrInUse => {
-                        // We hit a used port, try again.
+                    ErrorKind::AddrInUse | ErrorKind::PermissionDenied => {
```
Comment (martinthomson) [inline src/lib.rs:344]: Did you pick up an extra change somewhere?

**Review by martinthomson — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (larseggert) [reply src/lib.rs:344]: Yes, this was something I hit while testing. See first post. Should I factor this out?

Comment (martinthomson) [reply src/lib.rs:344]: I don't mind; hence the approval.  But it's going to be hard to identify, so maybe that is the right thing to do.

Comment (larseggert) [reply src/lib.rs]: I split this out into #24

**Pull request merged**